pkg/debug/checkups/processes.go

```diff
@@ -24,32 +25,44 @@ func (c *Processes) Run(ctx context.Context, fullWriter io.Writer) error {
 		return errors.New("missing io.Writer")
 	}
 
-	c.data = []string{}
+	c.data = []map[string]any{}
 
 	ps, err := process.ProcessesWithContext(ctx)
 	if err != nil {
 		return fmt.Errorf("getting process list: %w", err)
 	}
 
-	for _, p := range ps {
-		exe, _ := p.Exe()
+	jsonWriter := json.NewEncoder(fullWriter)
 
-		// full gets the full process logs. This lets us examine them for processes that might interfere.
-		fmt.Fprintf(fullWriter, p.String())
+	for _, p := range ps {
+		// This doesn't feel great yet. I'm not sure what data we need, and some of the gopsutil stuff has
+		// weird extraneous exec routines. So this is a starting point
+		pMap := map[string]any{
+			"pid":         p.Pid,
+			"exe":         naIfError(p.ExeWithContext(ctx)),
+			"cmdline":     naIfError(p.CmdlineSliceWithContext(ctx)),
+			"create_time": naIfError(p.CreateTimeWithContext(ctx)),
+			"ppid":        naIfError(p.PpidWithContext(ctx)),
+			"mem_info":    naIfError(p.MemoryInfoWithContext(ctx)),
+			"mem_info_ex": naIfError(p.MemoryInfoExWithContext(ctx)),
+			"cpu_times":   naIfError(p.TimesWithContext(ctx)),
+			"status":      naIfError(p.StatusWithContext(ctx)),
+			"uids":        naIfError(p.UidsWithContext(ctx)),
+		}
```

Comment:
Why are we running these for all processes if we're only going to save the results for Kolide processes?

Reply:
We include all of it in the extra data. Line 52: `_ = jsonWriter.Encode(pMap)`

We don't put it all into `c.data` because that will end up in the logs.